ksql-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenSpec.java

```diff
@@ -102,11 +114,18 @@ void addFunction(FunctionName functionName, Kudf function) {
       argumentBuilder.add(new FunctionArgumentSpec(codeName, function.getClass(), function));
     }
 
+    void addSchema(StructExpression struct, Schema schema) {
+      final String schemaName = CodeGenUtil.schemaName(schemaCount++);
```

Comment:
do you need the counter? Can you not just use the structToSchemaName size?

Reply:
After changing this to use `ImmutableMap.Builder`, you can't get the size from the builder. I have kept this in favor of using ImmutableMap.